src/System.Private.Xml/src/System/Xml/Core/XmlWellFormedWriter.cs

```diff
@@ -634,6 +634,15 @@ public override void WriteStartAttribute(string prefix, string localName, string
                         throw new ArgumentException(SR.Xml_EmptyLocalName);
                     }
                 }
+                // check local name
+                if (prefix == null || prefix.Length == 0)
+                {
+                    if (localName.Contains("xmlns:xs"))
```

Comment:
I don't think this is the right place to solve this problem - if we are gonna be adding this convenience then the other argument should be a default value = null and first one we split by semicolon and not just special casing for "xmlns:xs".

Reply:
I think the fix should be wherever we call call WriteStartAttribute - if this is done by XML implementation please reduce the repro and file an issue on that.